### ISSUE TITLE
Not an editor command: PackerSync

### ISSUE THREAD

**Issue by CMOISDEAD** [wbthomason/packer.nvim#834]
- `nvim --version`: 0.6.1
- `git --version`:2.35.1

When executing the Packer commands it says that it is not an editor command.
help

**Comment by sdondley** [comment]
Works for me. Try deleting the plugin/packer_compiled.lus file. Then open new file and recompile with :PackerCompile

Also, per help docs, make sure you have `require('packer').startup(...)` in your config.

**Comment by dongwangdw** [comment]
I got the same issue built from the latest code.
OS: Mac
git: 2.32.0

```
NVIM v0.7.0-dev+1193-g72a04bda9
Build type: Release
LuaJIT 2.1.0-beta3
```

**Comment by Guzzii** [comment]
I am having the same issue. None of packer commands are available. only `packadd` and `packloadall` is there. Anyone could give a direction for debug? This renders my neovim unusable

To reproduce:
```bash
$ git clone --depth 1 https://github.com/wbthomason/packer.nvim\
 ~/.local/share/nvim/site/pack/packer/start/packer.nvim
```
init.lua
```lua
return require('packer').startup(function(use)
  -- Packer can manage itself
  use 'wbthomason/packer.nvim'
end)
```

**Comment by leonardchinonso** [comment]
I'm currently having the same issue

**Comment by bryant-the-coder** [comment]
Upgrade your neovim to 0.7 (latest release)

**Comment by dongwangdw** [comment]
Atually I got this work by installing the packer.vim and initializing the lua script.

**Comment by iDuuck** [comment]
Jep, just initialized a new computer with my dotfiles, and this is the error I am seeing.

**Comment by Chris56974** [comment]
I fixed this by adding the [packer.nvim bootstrap](https://github.com/wbthomason/packer.nvim#bootstrapping) part to my plugin script.

**Comment by burritoOverflow** [comment]
This happens to me somewhat regularly using `NVIM v0.9.0-dev-1222+gf01f18cdf`.

Navigating to the script containing packer install function and sourcing it works for me.

**Comment by danlikestocode** [comment]
> This happens to me somewhat regularly using `NVIM v0.9.0-dev-1222+gf01f18cdf`.
> 
> Navigating to the script containing packer install function and sourcing it works for me.

I am also having this issue on `v0.9.0-dev-1237+gaccdb0104`.

Same solution as well, navigate to packer.lua file containing packer install function and sourcing it...

**Comment by shalomb** [comment]
Also on ` NVIM v0.9.0-dev-1288+g4863ca6b8`

Sourcing `packer.lua` doesn't cut it for me. :|

**Comment by nfjdu** [comment]
![image](https://user-images.githubusercontent.com/43147143/230640036-b40d22e4-56ae-4e1c-a221-f8f71d03c82b.png)
Installed nvim using newest instructions. Version `NVIM v0.9.0` installed. All other my vim configs is loaded and working.. Only problem is packer: `Not an editor command: PackerSync `. Installed packer as usual (off docs) `git clone --depth 1 https://github.com/wbthomason/packer.nvim\
 ~/.local/share/nvim/site/pack/packer/start/packer.nvim`. Any suggestions?

**Comment by yunxiaoxiao11** [comment]
exit wsl -> cd your plugins's dir -> vim or nvim plugins.lua
now I can excute PackerSync command

**Comment by bvidosits** [comment]
same issue on macos, had the bootstrap code in there as well but opening the packer file resulted in errors (from the plugins I didn't have installed yet but also `PackerSync` not being a command) and then an empty file.

Solution was to go full caveman and repeatedly try to open the packer file, scroll through the errors instead of `q`, eventually the file loaded but `:so` was still broke, one last reopen made `:so` work and I could finally `PackerSync`

tl;dr: Keep hitting it with a hammer and it'll eventually work

**Comment by lavantien** [comment]
- Same issue: Cannot run `:PackerSync` unless I go into the [`packer.lua`](https://github.com/lavantien/nvim/blob/main/lua/lavantien/packer.lua) file and execute `:so` first

- Neovim version:

```
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/home/linuxbrew/.linuxbrew/Cellar/neovim/0.9.0/share/nvim"
```

- Checkhealth result:

<details>
  <summary>expand</summary>
  
```
==============================================================================
mason: require("mason.health").check()

mason.nvim ~
- OK mason.nvim version v1.1.0
- OK PATH: prepend
- OK Providers: 
  mason.providers.registry-api
  mason.providers.client
- OK neovim version >= 0.7.0

mason.nvim [Registries] ~
- OK Registry `github.com/mason-org/mason-registry version: 2023-05-23-lively-flame` is installed.

mason.nvim [Core utils] ~
- OK unzip: `UnZip 6.00 of 20 April 2009, by Debian. Original by Info-ZIP.`
- OK wget: `GNU Wget 1.21.2 built on linux-gnu.`
- OK curl: `curl 8.1.1 (x86_64-pc-linux-gnu) libcurl/8.1.1 OpenSSL/1.1.1t zlib/1.2.13 brotli/1.0.9 zstd/1.5.5 libidn2/2.3.4 libssh2/1.10.0 nghttp2/1.53.0 librtmp/2.3`
- OK gzip: `gzip 1.10`
- OK tar: `tar (GNU tar) 1.34`
- OK bash: `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
- OK sh: `Ok`

mason.nvim [Languages] ~
- OK Go: `go version go1.20.4 linux/amd64`
- OK Ruby: `ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux]`
- OK PHP: `PHP 8.2.6 (cli) (built: May  9 2023 06:25:31) (NTS)`
- OK cargo: `cargo 1.69.0 (6e9a83356 2023-04-12)`
- OK node: `v20.2.0`
- OK luarocks: `/home/linuxbrew/.linuxbrew/bin/luarocks 3.9.2`
- OK Composer: `Composer version 2.5.5 2023-03-21 11:50:05`
- OK julia: `julia version 1.9.0`
- OK java: `openjdk version "20.0.1" 2023-04-18`
- OK python3: `Python 3.11.3`
- OK RubyGem: `3.4.10`
- OK javac: `javac 20.0.1`
- OK npm: `9.6.6`
- OK pip3: `pip 23.1.2 from /home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.3/lib/python3.11/site-packages/pip (python 3.11)`

mason.nvim [GitHub] ~
- OK GitHub API rate limit. Used: 3. Remaining: 4997. Limit: 5000. Reset: Wed 24 May 2023 10:38:17 AM +07.

==============================================================================
null-ls: require("null-ls.health").check()

- OK gofumpt: the command "gofumpt" is executable.
- OK goimports_reviser: the command "goimports-reviser" is executable.
- OK dart_format: the command "dart" is executable.
- OK deno_fmt: the command "deno" is executable.
- OK rustfmt: the command "rustfmt" is executable.
- OK autopep8: the command "autopep8" is executable.
- OK zigfmt: the command "zig" is executable.
- OK htmlbeautifier: the command "htmlbeautifier" is executable.
- OK xmlformat: the command "xmlformat" is executable.
- OK stylua: the command "stylua" is executable.
- OK yamlfmt: the command "yamlfmt" is executable.
- OK shfmt: the command "shfmt" is executable.
- OK sqlfmt: the command "sqlfmt" is executable.
- OK nixfmt: the command "nixfmt" is executable.
- OK nginx_beautifier: the command "nginxbeautifier" is executable.
- OK prettier: the command "prettier" is executable.
- OK clang_format: the command "clang-format" is executable.
- OK markdownlint: the command "markdownlint" is executable.
- OK eslint: the command "eslint" is executable.
- OK golangci_lint: the command "golangci-lint" is executable.
- OK protolint: the command "protolint" is executable.
- OK hadolint: the command "hadolint" is executable.
- OK yamllint: the command "yamllint" is executable.
- OK buf_lint: the command "buf" is executable.
- OK markdownlint: the command "markdownlint" is executable.
- OK terraform_validate: the command "terraform" is executable.
- OK ansiblelint: the command "ansible-lint" is executable.
- OK actionlint: the command "actionlint" is executable.
- OK dotenv-linter: the command "dotenv-linter" is executable.
- OK checkmake: the command "checkmake" is executable.
- OK djlint: the command "djlint" is executable.
- OK eslint: the command "eslint" is executable.
- OK flake8: the command "flake8" is executable.
- OK clang_check: the command "clang-check" is executable.
- OK codespell: the command "codespell" is executable.
- OK gomodifytags: the source "gomodifytags" can be ran.
- OK impl: the source "impl" can be ran.
- OK eslint: the command "eslint" is executable.
- refactoring: cannot verify if the command is an executable.
- OK shellcheck: the command "shellcheck" is executable.
- OK gitsigns: the source "gitsigns" can be ran.

==============================================================================
nvim: require("nvim.health").check()

Configuration ~
- OK no issues found

Runtime ~
- OK $VIMRUNTIME: /home/linuxbrew/.linuxbrew/Cellar/neovim/0.9.0/share/nvim/runtime

Performance ~
- OK Build type: Release

Remote Plugins ~
- OK Up to date

terminal ~
- key_backspace (kbs) terminfo entry: `key_backspace=^H`
- key_dc (kdch1) terminfo entry: `key_dc=\E[3~`
- $TERM_PROGRAM="WezTerm"
- $COLORTERM="truecolor"

==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- OK `tree-sitter` found 0.20.8 (d4c1bf7ce78051b7f4a381d1508d68928512ed5f) (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v20.2.0 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: cc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "5.19.0-42-generic",
  sysname = "Linux",
  version = "#43~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Apr 21 16:51:08 UTC 2"
} ~

Parser/Features         H L F I J
  - bash                ✓ ✓ ✓ . ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - css                 ✓ . ✓ ✓ ✓
  - diff                ✓ . . . .
  - dockerfile          ✓ . . . ✓
  - gitcommit           ✓ . . . ✓
  - gitignore           ✓ . . . .
  - go                  ✓ ✓ ✓ ✓ ✓
  - gomod               ✓ . . . ✓
  - gosum               ✓ . . . .
  - gowork              ✓ . . . ✓
  - graphql             ✓ . . ✓ ✓
  - html                ✓ ✓ ✓ ✓ ✓
  - http                ✓ . . . ✓
  - java                ✓ ✓ ✓ ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - jsdoc               ✓ . . . .
  - json                ✓ ✓ ✓ ✓ .
  - lua                 ✓ ✓ ✓ ✓ ✓
  - make                ✓ . ✓ . ✓
  - markdown            ✓ . ✓ . ✓
  - nix                 ✓ ✓ ✓ . ✓
  - proto               ✓ . ✓ . .
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - rust                ✓ ✓ ✓ ✓ ✓
  - scss                ✓ . ✓ ✓ .
  - toml                ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓
  - yaml                ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

==============================================================================
provider: health#provider#check

Clipboard (optional) ~
- OK Clipboard tool found: xclip

Python 3 provider (optional) ~
- `g:python3_host_prog` is not set.  Searching for python3 in the environment.
- Multiple python3 executables found.  Set `g:python3_host_prog` to avoid surprises.
- Executable: /home/linuxbrew/.linuxbrew/bin/python3
- Other python executable: /usr/bin/python3
- Other python executable: /bin/python3
- Python version: 3.11.3
- pynvim version: 0.4.3
- OK Latest pynvim is installed.

Python virtualenv ~
- OK no $VIRTUAL_ENV

Ruby provider (optional) ~
- Ruby: ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux]
- Host: /home/linuxbrew/.linuxbrew/lib/ruby/gems/3.2.0/bin/neovim-ruby-host
- OK Latest "neovim" gem is installed: 0.9.0

Node.js provider (optional) ~
- Node.js: v20.2.0
- Nvim node.js host: /home/linuxbrew/.linuxbrew/lib/node_modules/neovim/bin/cli.js
- OK Latest "neovim" npm/yarn/pnpm package is installed: 4.10.1

Perl provider (optional) ~
- perl executable: /home/linuxbrew/.linuxbrew/bin/perl
- OK Latest "Neovim::Ext" cpan module is installed: 0.06

==============================================================================
telescope: require("telescope.health").check()

Checking for required plugins ~
- OK plenary installed.
- OK nvim-treesitter installed.

Checking external dependencies ~
- OK rg: found ripgrep 13.0.0
- OK fd: found fd 8.7.0

===== Installed extensions ===== ~

==============================================================================
vim.lsp: require("vim.lsp.health").check()

- LSP log level : WARN
- Log path: /home/lavantien/.local/state/nvim/lsp.log
- Log size: 2511 KB

vim.lsp: Active Clients ~
- No active clients

==============================================================================
vim.treesitter: require("vim.treesitter.health").check()

- Nvim runtime ABI version: 14
- OK Parser: bash       ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/bash.so
- OK Parser: c          ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/c.so
- OK Parser: css        ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/css.so
- OK Parser: diff       ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/diff.so
- OK Parser: dockerfile ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/dockerfile.so
- OK Parser: gitcommit  ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/gitcommit.so
- OK Parser: gitignore  ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/gitignore.so
- OK Parser: go         ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/go.so
- OK Parser: gomod      ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/gomod.so
- OK Parser: gosum      ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/gosum.so
- OK Parser: gowork     ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/gowork.so
- OK Parser: graphql    ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/graphql.so
- OK Parser: html       ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/html.so
- OK Parser: http       ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/http.so
- OK Parser: java       ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/java.so
- OK Parser: javascript ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/javascript.so
- OK Parser: jsdoc      ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/jsdoc.so
- OK Parser: json       ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/json.so
- OK Parser: lua        ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/lua.so
- OK Parser: make       ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/make.so
- OK Parser: markdown   ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/markdown.so
- OK Parser: nix        ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/nix.so
- OK Parser: proto      ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/proto.so
- OK Parser: python     ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/python.so
- OK Parser: query      ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/query.so
- OK Parser: rust       ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/rust.so
- OK Parser: scss       ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/scss.so
- OK Parser: toml       ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/toml.so
- OK Parser: typescript ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/typescript.so
- OK Parser: vim        ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/vim.so
- OK Parser: vimdoc     ABI: 14, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/vimdoc.so
- OK Parser: yaml       ABI: 13, path: /home/lavantien/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/yaml.so
- OK Parser: c          ABI: 14, path: /home/linuxbrew/.linuxbrew/Cellar/neovim/0.9.0/lib/nvim/parser/c.so
- OK Parser: lua        ABI: 14, path: /home/linuxbrew/.linuxbrew/Cellar/neovim/0.9.0/lib/nvim/parser/lua.so
- OK Parser: query      ABI: 14, path: /home/linuxbrew/.linuxbrew/Cellar/neovim/0.9.0/lib/nvim/parser/query.so
- OK Parser: vim        ABI: 14, path: /home/linuxbrew/.linuxbrew/Cellar/neovim/0.9.0/lib/nvim/parser/vim.so
- OK Parser: vimdoc     ABI: 14, path: /home/linuxbrew/.linuxbrew/Cellar/neovim/0.9.0/lib/nvim/parser/vimdoc.so
```

</details>

**Comment by mauke** [comment]
I fixed it by adding
```vim
lua require('plugins')
```
to `~/.config/nvim/init.vim`.

**Comment by LeafMunchingQuoka** [comment]
For anyone who's been following tutorial by ThePrimeagen: you need to add `require("<your-folder-name>.packer")` inside your `$HOME/.config/nvim/lua/<your-folder>/init.lua`. <your-folder> is the name of the directory you created in .config/nvim/lua. In the [tutorial](https://youtu.be/w7i4amO_zaE) he names it "theprimeagen".

**Comment by SimbaNinja917** [comment]
> For anyone who's been following tutorial by ThePrimeagen: you need to add `require("<your-folder-name>.packer")` inside your `$HOME/.config/nvim/lua/<your-folder>/init.lua`. your-folder is the name of the directory you created in .config/nvim/lua. In the [tutorial](https://youtu.be/w7i4amO_zaE) he names it "theprimeagen".

solved it for me! thanks !

**Comment by talipkorkmazer** [comment]
I was also following the tutorial by ThePrimeagen. For me, I had an empty space inside my packer.lua file. The LSP was complaining about it when I opened it. When I removed the empty spaces, :PackerSync worked.

**Comment by sir-george2500** [comment]
I was experiencing this same issues my solution was 

I remove the packer_compiled 
`rm -r  plugin/packer_compiled.lua`

then I when back into my `packer.lua` or where you are installing your plugin 

I source it using `:so` 

after that I ran the command `:PackerSync` and it work

**Comment by BrPrS** [comment]
> I was experiencing this same issues my solution was
> 
> I remove the packer_compiled `rm -r plugin/packer_compiled.lua`
> 
> then I when back into my `packer.lua` or where you are installing your plugin
> 
> I source it using `:so`
> 
> after that I ran the command `:PackerSync` and it work

this solution worked for me

**Comment by zacck** [comment]
> For anyone who's been following tutorial by ThePrimeagen: you need to add `require("<your-folder-name>.packer")` inside your `$HOME/.config/nvim/lua/<your-folder>/init.lua`. your-folder is the name of the directory you created in .config/nvim/lua. In the [tutorial](https://youtu.be/w7i4amO_zaE) he names it "theprimeagen".

Fixed it for me!

**Comment by deletec00kiesb4leaving** [comment]
Thank you @LeafMunchingQuoka It worked <3
